tests/c/outline_recursion.newcg.c

```diff
@@ -0,0 +1,75 @@
+// ignore-if: test $YK_JIT_COMPILER != "yk" -o "$YKB_TRACER" = "swt"
+// Run-time:
+//   env-var: YKD_LOG_IR=-:aot,jit-pre-opt
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_LOG_JITSTATE=-
+//   stderr:
+//     jitstate: start-tracing
+//     0
+//     1
+//     2
+//     3
+//     4
+//     jitstate: stop-tracing
+//     --- Begin aot ---
+//     ...
+//     func main($arg0: i32, $arg1: ptr) -> i32 {
+//     ...
+//     --- End aot ---
+//     --- Begin jit-pre-opt ---
+//     ...
+//     Call @foo(%{{4}})
+//     ...
+//     --- End jit-pre-opt ---
+//     0
+//     1
+//     2
+//     3
+//     jitstate: enter-jit-code
+//     0
+//     1
+//     2
+//     0
+//     1
+//     jitstate: deoptimise
+//     0
+//     exit
+
+// Test outlining of recursive calls.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+void foo(int i) {
```

Comment:
might be worth adding `__attribute__((noiline))` (or whatever it is) to ensure this doesn't get inlined when we do more AOT optimistations?

Reply:
Then it will just fail the test. But sure I can add it.

Reply:
[84b791a](https://github.com/ykjit/yk/pull/1121/commits/84b791a02b0b351cf8ac2525757f21ccc2344450)